docs/exercises/fix-errors.md

```diff
@@ -191,6 +191,93 @@
+        Color color; //Objektattribut, wird im Konstruktor Car initialisiert
+
+        public Car(Color color){
+            this.color = color;
+        }
+        public static int noOfWheels(){ //Klassenmethode
+            return noOfWheels;
+        }
+        public Color getColor(){ //Objektmethode
+            return color;
+        }
+    }
+    ````
+    ```` java linenums="1"
+    //in Main
+    Car myCar = new Car((Color.black));
+    Color myCarColor = myCar.getColor();  //korrekt: Objektmethode mit Objekt aufrufen
+    int myCarWheels = myCar.noOfWheels(); //inkorrekt: Klassenmethode mit Objekt aufrufen
+    int carWheels = Car.noOfWheels();     //korrekt: Klassemnethode mit Klassennamen aufrufen
+    ````
+
+    Bei Zeile 4 in Main wird einem Folgendes angezeigt:
+    ````
+    Static member 'Car.noOfWheels()' accessed via instance reference 
+    ````
+    Statt die Klassenmethode mit einem Objekt der Klasse aufzurufen, ruft man sie einfach mit dem Namen der Klasse auf.
+    Mehr zu Klassenmethoden kann im Foliensatz 4c) auf den Seiten 34-50 gefunden werden.
+
+    ### isBooleanExpression == false
+
+    In der Bedingung einer if-Verzweigung, der Fortsetzungsbedingung einer while-Schleife und an anderen Stellen verwendet man boolesche Ausdrücke. Diese können zu true oder zu false auswerten. Außer man verwendet keinen booleschen Ausdruck, denn dann kompiliert der Code nicht.
+
+    Was sind also alles boolesche Ausdrücke und was nicht?  
+    Die booleschen Werte true und false sind die einfachsten booleschen Ausdrücke. Diese können z.B. mit dem logischen Oder (||) bzw. dem logischen Und (&&) oder der Negation (!) zu komplizierteren Ausdrücken kombiniert werden. Vergleiche von Zahlen mit ==, <, >=, != etc. funktionieren auch. 
+
+    Was sind häufig verwendete, aber inkorrekte Ausdrücke?  
+    Oft wird beim Vergleich von z.B. zwei Zahlen nur ein = verwendet, gemeint ist meist der Vergleich mit ==.  
+    Methoden, die keinen boolean zurückgeben eignen sich nicht direkt, außer man nutzt die Rückgabe und z.B. die equals-Methode, um einen booleschen Ausdruck zu erzeugen.   
+    Nicht ganz falsch, aber etwas unnötig, ist es, einen boolean mit true oder false zu vergleichen: isOdd(5)==true ist semantisch äquivalent zu isOdd(5). Auch isEven(4) == false kann einfach zu !isEven(4) vereinfacht werden.
+
+    Auch hier nochmal ein paar Beispiele:
+
+    ```` java linenums="1"
+    int a = 5;
+    if (a = 4){...} // hier wäre == richtig, denn eine Zuweisung ist kein boolescher Ausdruck
+
+    while( isLessThan7(a) == false){ //ist zwar möglich, aber !lessThan7(a) wäre schöner
+        a++;
+    }
+
+    boolean b = myCar.getColor(); 
+    //falsch, da getColor eine Farbe und keinen boolean zurückgibt
+
+    boolean c = myCar.getColor().equals(Color.green); 
+    //korrekt, die equals Methode vergleicht die beiden Farben und gibt einen boolean zurück
+    ````
```

Comment:
Maybe show the following case as a reason to not use bool == true:

boolean a = false;
if (a = true) {
    System.out.println("a is true");
}

It will print "a is true" because the assignment operator returns the assigned value.